### PR TITLE
Documentation cleanup

### PR DESCRIPTION
**Pull request by robertzk**

### CODE REVIEW

File: R/hooks.R [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#' Add a hook to a tundraContainer.
+#'
+#' Hooks are useful for defining additional checks that should be
+#' performed prior to and during training and prediction. For example,
+#' one might want to issue a warning if the user is predicting on 
+#' rows that were used for training, or a sanity check might be 
+#' present prior to training to ensure a dependent variable is present.
```
Comment (peterhurford) [inline R/hooks.R]: Awesome!

File: R/tundraContainer-train.R [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#'    provided to the container during initialization will be assumed
+#'    to have been trained, and the \code{dataframe} provided will not
+#'    be run through it.
+#' @return The value returned by the \code{tundraContainer}'s
+#'    \code{train_function}. Since the \code{train_function} has side effects
+#'    on the container as its primary purpose, this can usually be
+#'    \code{invisible(NULL)}.
+train <- function(dataframe, train_args = list(), verbose = FALSE, munge = TRUE) {
+  if (isTRUE(self$.trained)) {
+    stop("The tundra ", sQuote(self$.keyword), " model has already been trained.")
```
Comment (peterhurford) [inline R/tundraContainer-train.R]: Why can't you retrain?

File: R/utils.r [hunk 3]
```diff
@@ -1,2 +1,72 @@
...
+    name <- names(list2)[i]
+    if (!identical(name, NULL) && !identical(name, "")) list1[[name]] <- list2[[i]]
+    else list1 <- append(list1, list(list2[[i]]))
+  }
+  list1
+}
+
+is.simple_string <- function(obj) {
+  is.character(obj) && length(obj) == 1 && !is.na(obj) && nzchar(obj)
+}
```
Comment (peterhurford) [inline R/utils.r]: Nice.

File: R/tundraContainer-train.R (hunk 2 above)
Comment (robertzk) [inline R/tundraContainer-train.R]: Well, it's meant to be used in production, so kind of a safety measure. I guess it's not strictly necessary!

**Pull request merged**